frontend/src/components/PasswordModal.js

```diff
@@ -57,10 +59,22 @@ function PasswordModal({state, dismiss, done, setPasswordContext}) {
         
         const salt = Utilities.base64ToArrayBuffer(passwordData["salt"])
         const aesKey = await PasswordEncryption.deriveAesKey(password, salt);
-        const keyHash = Utilities.arrayBufferToBase64(await PasswordEncryption.getKeyHash(aesKey));
 
-        if(passwordData.hash !== keyHash) {
-            showAlert("Incorrect Password!");
+        try {
+            // AES-GCM has verification built in. Should detect if the key is wrong.
+            let contracts = await Contracts.load(aesKey);
+            let accounts = await Accounts.load(aesKey);
+
+            if(contracts === null && accounts === null) {
+                // We have no data to verify if the password is correct...
```

Comment:
Could we just throw an error here or remove it? IT seems a bit redundant...

Reply:
Well, currently i dont know how we can verify it. 
Throwing an error will break the application for users that create a password but do not have any other stored information. This could be more common than we think.